web_client/src/App.js

```diff
@@ -0,0 +1,48 @@
+import React, { Component, Fragment } from 'react';
+import { Route } from 'react-router-dom';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import { ConnectedRouter } from 'connected-react-router';
+import { handleApplicationStarted } from 'actions/app';
+import { getLoading } from 'selectors';
+import Loading from 'scenes/Loading';
+import Landing from 'scenes/Landing';
+import Playlists from 'scenes/Playlists';
+import SouffleStation from 'scenes/SouffleStation';
+
+const App = ({ history, loading }) => (
+  loading ? <Loading /> :
+  <ConnectedRouter history={history}>
+    <Fragment>
+      <Route path='/' exact component={Landing}/>
+      <Route path='/playlists' exact component={Playlists}/>
+      <Route path='/playlists/:id' component={SouffleStation}/>
+    </Fragment>
+  </ConnectedRouter>
+);
+
+class AppContainer extends Component {
+  componentDidMount = () => {
+    this.props.onApplicationStarted();
+  }
+
+  render = () => (
+    <App {...this.props} />
+  )
+}
+
+const mapStateToProps = (state, props) => ({
+  ...props,
+  loading: getLoading(state)
+});
+
+const mapDispatchToProps = dispatch => {
+  return bindActionCreators(
```

Comment:
you don't need `bindActionCreators`. You get it for free with `react-redux`: https://github.com/reduxjs/react-redux/blob/master/docs/api.md#connectmapstatetoprops-mapdispatchtoprops-mergeprops-options

Reply:
ooo cool. i hate `bindActionCreators` anyway.

Reply:
and it means all your `mapDispatchToProps` can just be dictionaries, and none of your components need to call dispatch. they can just call the function.

Also, why are you renaming the `on` prefixed function to `handle`?

Reply:
https://redux-form.com/7.4.2/docs/faq/handlevson.md/

Reply:
👍 (that seems like an unnecessary distinction, but i get it. and it seems like a standard.)